kornia/color/zca.py

```diff
@@ -0,0 +1,208 @@
+from typing import Tuple
+from functools import reduce
+
+import torch
+import torch.nn as nn
+
+
+class ZCAWhitening(nn.Module):
```

Comment:
in my opinion, and for consistency to what we have been following, nn.Module should just be a wrapper around free functions that makes easy to be included nn.Sequential.

If possible I would like to have a free functional with the following signature:
```python
def zca_whitten(input: torch.Tensor, mean: Optional[torch.Tensor] = None, transform: Optional[torch.Tensor] = None) -> torch.Tensor:
    if mean is None and transform is None:
       transform, mean = zca_mean(input)
   # then apply
```
I'm not a huge fan of state functions, at least from the point of view of low level libs like this. I think the use can do it by itself. However, I'm totally open to iterate this proposal towards this direction.

Reply:
Yeah I agree. I think this was my number one point of contention when implementing this.